crates/solvers/src/domain/solver/baseline.rs

```diff
@@ -126,10 +126,30 @@ impl Inner {
             boundary::baseline::Solver::new(&self.weth, &self.base_tokens, &auction.liquidity);
 
         for (i, order) in auction.orders.into_iter().enumerate() {
-            let sell_token = auction.tokens.reference_price(&order.sell.token);
             let Some(user_order) = UserOrder::new(&order) else {
                 continue;
             };
+
+            let sell_token = match auction.tokens.reference_price(&order.sell.token) {
```

Comment:
Still quite hard to read IMO.
I see that using early returns here doesn't work because we need to return from the block. And a separate function might need a bunch of arguments and still needs to have the special case for the `continue`.

I tried to play a bit of code golf to see if I could reduce the indentation somewhat.
Feel free to use if you like it or ignore if you think it doesn't improve readability.


```rust
            let sell_token = user_order.get().sell.token;
            let sell_token_price = match auction.tokens.reference_price(&sell_token) {
                Some(price) => price,
                None if sell_token == self.weth.0.into() => {
                    // Early return if the sell token is native token
                    auction::Price(eth::Ether(eth::U256::exp10(18)))
                }
                None => {
                    // Estimate the price of the sell token in the native token
                    let native_price_request = self.native_price_request(user_order);
                    if let Some(route) = boundary_solver.route(native_price_request, self.max_hops) {
                        // how many units of buy_token are bought for one unit of sell_token
                        // (buy_amount / sell_amount).
                        let price = self.native_token_price_estimation_amount.to_f64_lossy()
                            / route.input().amount.to_f64_lossy();
                        let Some(price) = to_normalized_price(price) else {
                            continue;
                        };

                        auction::Price(eth::Ether(price))
                    } else {
                        // This is to allow quotes to be generated for tokens for which the sell
                        // token price is not available, so we default to fee=0
                        auction::Price(eth::Ether(eth::U256::MAX))
                    }
                }
            };
```